src/components/Graph/ThresholdLabel.js

```diff
@@ -1,24 +1,54 @@
-import React from 'react';
+import React, { Component } from 'react';
+import TooltipHandler from '../Filters/TooltipHandler';
 import { addCommas, getReadableDate } from '../../utils/utils.js';
 
-function ThresholdLabel(props) {
-    const { r0full, r0selected } = props;
-    const chance = Math.round(100 * props.percExceedence);
-    const val = addCommas(Math.ceil(props.statThreshold / 100) * 100);
-    const date = getReadableDate(props.dateThreshold);
+class ThresholdLabel extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            showTooltip: false
+        }
+    }
+
+    handleTooltipClick = () => {
+        this.setState({showTooltip: !this.state.showTooltip})
```

Comment:
This is more of a tip:

`handleTooltipClick = () => this.setState({showTooltip: !this.state.showTooltip})` can cause unwanted side effects if the user rapidly clicks on the tool tip. For example, if there is an API call attached to the press, pressing the button rapidly could result in multiple API calls.

A way to get around this is to be explicit about what is happening to the state
`showToolTip = () => this.setState({showTooltip: true})`
`hideToolTip = () => this.setState({showTooltip: false})`

If done this way, there might be more setup but it allows the tooltip to fully render before it allows it to be closed.